logging/log.go

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright (c) 2018 Juniper Networks, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Function doesn't return error, because it is just for logging.
+// If conversion to json returns error we want to log variable as raw
+func VariableToJSON(variable interface{}) string {
+	jsonOutput, err := json.Marshal(variable)
+	if err != nil {
+		log.Debugln("Converting to JSON error:", err)
+		return fmt.Sprintf("Cannot convert request to JSON. Raw output: %s", variable)
+	}
+	return string(jsonOutput)
+}
+
+func HTTPMessageLogger(param interface{}) string {
+
+	if param == nil {
+		return "Empty request/response. Some error occured."
```

Comment:
Why are we using `log.Debugln("")` in other places for error logging but not here?

Should we `log.Debugln("")` in other places anyways? Such message would show up in debug-level log anyways in the http log message.

Reply:
Because we don't know if it is an error.

Reply:
Hmm, so maybe let's add this clarification in the code